rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/activityList.additional.spec.ts

```diff
@@ -0,0 +1,244 @@
+    const spy = jest
+      .spyOn(wrapper.vm, "showBulkEditResults", "get")
+      .mockReturnValue(true);
+
+    const bulkDeleteResults = wrapper.find(
+      '[data-test-id="modal-bulk-delete-results"]'
+    );
+    expect(bulkDeleteResults.exists()).toBe(true);
+    // Restore the original implementation
+    spy.mockRestore();
+  });
+});
+it("enables bulk delete button when items are selected", async () => {
+  const wrapper = await shallowMountActivityList();
+  wrapper.vm.reports = [
+    {
+      execution: {
+        id: "2",
+        permalink: "/project/test/execution/show/2",
+      },
+      status: "succeeded",
+      dateCompleted: "2024-05-22T14:33:52Z",
+      node: {
+        total: 1,
+        succeeded: 1,
+        failed: 0,
+      },
+    },
+  ];
+  wrapper.vm.bulkEditMode = true;
+  wrapper.vm.bulkSelectedIds = ["2"];
+  await wrapper.vm.$nextTick();
+  expect(wrapper.vm.bulkSelectedIds.length).toBeGreaterThan(0);
+});
+describe("ActivityList Miscellaneous", () => {
+  it("toggles auto-refresh", async () => {
+    const wrapper = await shallowMountActivityList();
+    const autoRefreshCheckbox = wrapper.find("#auto-refresh");
+    await autoRefreshCheckbox.setValue(true);
+    expect(wrapper.vm.autorefresh).toBe(true);
+    await autoRefreshCheckbox.setValue(false);
+    expect(wrapper.vm.autorefresh).toBe(false);
+  });
+  it("handles pagination correctly", async () => {
+    const wrapper = await shallowMountActivityList();
+    const paginationSpy = jest.spyOn(wrapper.vm, "changePageOffset");
+    const pagination = wrapper.findComponent({ name: "OffsetPagination" });
+    pagination.vm.$emit("change", 10);
+    await wrapper.vm.$nextTick();
+    expect(paginationSpy).toHaveBeenCalledWith(10);
+    paginationSpy.mockRestore();
+  });
+  it("navigates to execution link", async () => {
+    const wrapper = await shallowMountActivityList();
+    const originalLocation = window.location;
+    const mockLocation = { href: "", assign: jest.fn() };
+    // @ts-ignore
+    delete window.location;
+    // @ts-ignore
+    window.location = mockLocation;
+    wrapper.vm.reports = [
+      {
+        execution: {
+          id: "1",
+          permalink: "http://localhost:4440/project/test/execution/show/1",
+        },
+        status: "succeeded",
+        dateCompleted: "2024-05-22T14:33:52Z",
+        node: { total: 1, succeeded: 1, failed: 0 },
+      },
+    ];
+    await wrapper.vm.$nextTick();
+    console.log("Reports data in wrapper:", wrapper.vm.reports);
+    const reportRowItem = wrapper.find('[data-test-id="report-row-item"]');
+    if (reportRowItem.exists()) {
+      await reportRowItem.trigger("click");
+      console.log("After clicking report row item:", wrapper.html());
+    }
+    // Assert that navigation function was called with expected URL
+    expect(mockLocation.assign).toHaveBeenCalledWith(
+      "http://localhost:4440/project/test/execution/show/1"
+    );
```

Comment:
There's no need to do the mockLocation, replace the window object nor to spy on assign. Clicking the row will make the window location change and then you can just assert it:

```
expect(window.location).toBe("http://localhost:4440/project/test/execution/show/1")
```